pkg/snapshot/restorer/restorer.go

```diff
@@ -887,3 +945,84 @@ func ErrorArrayToError(errs []error) error {
 
 	return fmt.Errorf("%s", strings.TrimSpace(errString))
 }
+
+// handleAlarm function handles alarm raised by backup-restore.
+func (r *Restorer) handleAlarm(stopHandleAlarmCh chan bool, dbSizeAlarmCh <-chan string, dbSizeDisAlarmCh chan bool, clientMaintenance client.MaintenanceCloser) {
+	r.logger.Info("Starting to handle an alarm...")
+	for {
+		select {
+		case <-stopHandleAlarmCh:
+			r.logger.Info("Closing handle alarm...")
+			return
+		case endPoint := <-dbSizeAlarmCh:
+			r.logger.Info("Received a dbsize alarm")
+			r.logger.Infof("Calling defrag on endpoint: [%v]", endPoint)
+			if err := func() error {
+				ctx, cancel := context.WithTimeout(context.TODO(), etcdDefragTimeout)
```

Comment:
```suggestion
				ctx, cancel := context.WithTimeout(context.Background(), etcdDefragTimeout)
```
It's better to use `context.Background()` if no parent context is passed to the function. Similar for other places where `context.TODO()` is used